packages/contracts/contracts/tests/PaymentCycle_Test.sol

```diff
@@ -0,0 +1,155 @@
+    User private borrower;
+    User private lender;
+
+    TellerV2 tellerV2;
+    uint256 marketId;
+
+    WethMock wethMock;
+    TestERC20Token daiMock;
+
+    function setup_beforeAll() public {
+        // Deploy test tokens
+        wethMock = new WethMock();
+        daiMock = new TestERC20Token("Dai", "DAI", 10000000);
+
+        // Deploy Escrow beacon
+        CollateralEscrowV1 escrowImplementation = new CollateralEscrowV1();
+        UpgradeableBeacon escrowBeacon = new UpgradeableBeacon(
+            address(escrowImplementation)
+        );
+
+        // Deploy protocol
+        tellerV2 = new TellerV2(address(0));
+
+        // Deploy MarketRegistry & ReputationManager
+        IMarketRegistry marketRegistry = IMarketRegistry(new MarketRegistry());
+        IReputationManager reputationManager = IReputationManager(
+            new ReputationManager()
+        );
+        reputationManager.initialize(address(tellerV2));
+
+        // Deploy Collateral manager
+        CollateralManager collateralManager = new CollateralManager();
+        collateralManager.initialize(address(escrowBeacon), address(tellerV2));
+
+        // Deploy LenderCommitmentForwarder
+        LenderCommitmentForwarder lenderCommitmentForwarder = new LenderCommitmentForwarder(
+                address(tellerV2),
+                address(marketRegistry)
+            );
+
+        address[] memory lendingTokens = new address[](2);
+        lendingTokens[0] = address(wethMock);
+        lendingTokens[1] = address(daiMock);
+
+        // Initialize protocol
+        tellerV2.initialize(
+            50,
+            address(marketRegistry),
+            address(reputationManager),
+            address(lenderCommitmentForwarder),
+            lendingTokens,
+            address(collateralManager)
+        );
+
+        // Instantiate users & balances
+        marketOwner = new User(tellerV2, wethMock);
+        borrower = new User(tellerV2, wethMock);
+        lender = new User(tellerV2, wethMock);
+
+        uint256 balance = 50000;
+        payable(address(borrower)).transfer(balance);
+        payable(address(lender)).transfer(balance * 10);
+        borrower.depositToWeth(balance);
+        lender.depositToWeth(balance * 10);
+
+        daiMock.transfer(address(lender), balance * 10);
+        daiMock.transfer(address(borrower), balance);
+        // Approve Teller V2 for the lender's dai
+        lender.addAllowance(address(daiMock), address(tellerV2), balance * 10);
+
+        // Create a market with a monthly payment cycle type
+        marketId = marketOwner.createMarket(
+            address(marketRegistry),
+            0,
+            7000,
+            5000,
+            500,
+            false,
+            false,
+            PaymentType.EMI,
+            PaymentCycleType.Monthly,
+            "uri://"
+        );
+
+        (uint256 cycleValue, ) = marketRegistry.getPaymentCycle(marketId);
+    }
+
+    function paymentCycleType_test() public {
```

Comment:
its unclear what it is that we are testing in this file

also I don't like how much setup there is to only test a couple things - if anything we should be able to mock values to avoid having to implement a huge setup